### PR TITLE
Make OAuth providers optional

### PR DESCRIPTION
**Pull request by darrenmeehan**
I’ve made the OAuth providers optional by checking if their environment variables are set.

I’d like to update the environment variable validation stricter by adding a check that at least one OAuth provider is setup.

I need to test these changes actually work.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+
+
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
```
Comment (darrenmeehan) [inline .devcontainer/devcontainer.json:3]: I didn’t mean to add this yet, but no harm adding it in

**Comment by darrenmeehan** [comment]
It seems to be working OK. The site builds fine, and loads only showing Github as a login option. 

Despite having my Github OAuth app's callback url set as https://meehan-dive-log.vercel.app/api/auth/callback/github

I end up here for some reason

https://meehan-dive-log.vercel.app/login?callbackUrl=https%3A%2F%2Fmeehan-dive-log.vercel.app%2Flogin%3FcallbackUrl%3Dhttps%253A%252F%252Fmeehan-dive-log.vercel.app%252Fuser&error=Callback

I believe this is related to my configuration, not my code changes.

**Comment by darrenmeehan** [comment]
Once I get time I'll look into debugging using this. I'm using a remote dev environment so I'm not sure if I'll easily be able to debug it. 

https://nextjs.org/docs/pages/building-your-application/configuring/debugging#debugging-with-vs-code

**Comment by darrenmeehan** [comment]
I tried logging in with the preview but it redirects me back to the main app.

**Comment by darrenmeehan** [comment]
I changed my Vercel deployment to what is in main and am still seeing the same issue.

I must be missing some other configuration step.

File: README.md [hunk 2]
```diff
@@ -70,6 +70,32 @@ npx prisma studio
...
+  
+  - [@mauriciabad](https://github.com/mauriciabad/)
+  
+  ### Backend server
+  
+  We use Vercel as the backend server:
+  
+  
+  GITHUB_CLIENT_SECRET: z.string(),
+
```
Comment (mauriciabad) [inline README.md]:
````suggestion
- `DISCORD_CLIENT_ID, `DISCORD_CLIENT_SECRET`.
- `GOOGLE_CLIENT_ID`, `GOOGLE_CLIENT_SECRET`.
- `FACEBOOK_CLIENT_ID`, `FACEBOOK_CLIENT_SECRET`.
- `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`.
````

**Review by mauriciabad — COMMENTED** [review]
Just a change in the README, beside that LGTM

**Comment by mauriciabad** [comment]
I logged in in https://meehan-dive-log.vercel.app and it is working

**Comment by darrenmeehan** [comment]
> I logged in in https://meehan-dive-log.vercel.app and it is working

I had to push the schema to the database. I've never used Vercel or Planet scale so I assumed there was some magic to do that but they're wasn't.

I'll add a README change for it once I'm not on my phone.

**Pull request merged**